test/entities/test_yaml_to_objects.py

```diff
@@ -0,0 +1,262 @@
+      service_urls:
+        - https://service.prod.com
+    - name: sandbox
+      dashboard_url: https://dashboard.datadog.sandbox.com
+      health_check_url: https://dashboard.datadog.sandbox.com
+      service_urls:
+        - https://service.sandbox.com
+    impact: profit
+    integrations:
+      pagerduty_service: pagerduty_service1234
+      sentry_project: sentry project 15234
+    lifecycle: production
+"""
+
+    mocker.patch(
+        "zoo.repos.tasks.get_entity_file_content", return_value=service_component
+    )
+    uut.update_project_from_entity_file(proj=repository_dict)
+    component_entity = Entity.objects.first()
+    assert Entity.objects.all().count() == 1
+    assert Service.objects.all().count() == 1
+    assert component_entity.kind == "component"
+    assert component_entity.name == "base"
+    assert component_entity.owner == "platform"
+    assert component_entity.library is None
+    assert component_entity.type == "service"
+    assert component_entity.links.all().count() == 2
+    assert component_entity.group
+    assert component_entity.service is not None
+    assert component_entity.service.environments.all().count() == 2
+
+
+def test_create_base_component_and_library(mocker, repository_factory):
+    repository = repository_factory(
+        id=1,
+        remote_id=11,
+        name="test_proj1",
+        owner="john_doe1",
+        url="https://github.co  m/john_doe1/test_proj1",
+        provider="github",
+    )
+    repository_dict = {"id": repository.remote_id, "provider": repository.provider}
+
+    library_component = """
+- apiVersion: v1alpha1
+  kind_: component
+  metadata:
+    name: base_lib
+    owner: platform
+    group:
+      product_owner: john
+      project_owner: doe
+      maintainers: []
+    description: This is my fancy component
+    tags: []
+    links:
+    - name: Datadog
+      url: https://dashboard.datadog.com
+      icon: poop
+    - name: Sentry
+      url: https://sentry.skypicker.com
+  spec:
+    type_: library
+    impact: profit
+    integrations:
+      sonarqube_project: sonarqube
+    lifecycle: production
+"""
+
+    mocker.patch(
+        "zoo.repos.tasks.get_entity_file_content", return_value=library_component
+    )
+    uut.update_project_from_entity_file(proj=repository_dict)
+    component_entity = Entity.objects.first()
+    assert Entity.objects.all().count() == 1
+    assert Library.objects.all().count() == 1
+    assert component_entity.kind == "component"
+    assert component_entity.name == "base_lib"
+    assert component_entity.owner == "platform"
+    assert component_entity.type == "library"
+    assert component_entity.links.all().count() == 2
+    assert component_entity.group
+    assert component_entity.service is None
+    assert component_entity.library is not None
+
+
+def test_create_multiple_components(mocker, repository_factory):
+    multiple_components = """
+- apiVersion: v1alpha1
```

Comment:
this shouldn't be a list
in yaml, having multiple areas separated by `---` means multiple documents inside the same yaml file, which should be enough for what we're trying to do here, and it will also be more consistent with [backstage](https://backstage.io/docs/features/software-catalog/descriptor-format#overall-shape-of-an-entity)

so...
```yaml
apiVersion: v1alpha1
kind_: component
metadata:
  name: base
# morestuff
---
apiVersion: v1alpha1
kind_: component
metadata:
  name: service
# morestuff
```

Reply:
or do we have any other reason for needing a list here? from what I've seen everywhere is read as `entity[0]` all the time

Reply:
🤦 and i was confused why the hell i have to use `entity[0]` ... of course you are correct and it should not be list...